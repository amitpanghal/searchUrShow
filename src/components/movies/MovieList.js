import React from "react";
import Movie from "./Movie";
import AddMovie from "../common/AddMovie";

const MovieList = React.forwardRef(({ movies, handleClick }, ref) => {
  return (
    <div className="d-flex my-2 flex-wrap movie-container justify-content-center">
      {movies.map((movie, i) => {
        if (movie.poster_path) {
          return (
            <div
              key={movie.id + i}
              className="movie-div mt-1 mx-3 hover-button"
            >
              <Movie
                movie={movie}
                isLastMovie={i === movies.length - 1}
                ref={ref}
              />
              <AddMovie movie={movie} handleClick={handleClick} />
            </div>
          );
        } else {
          return null;
        }
      })}
    </div>
  );
});
export default MovieList;
