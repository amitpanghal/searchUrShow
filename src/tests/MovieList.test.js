import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import MovieList from "../components/movies/MovieList";

const renderMovieProps = args => {
  const defaultArgs = {
    movies: [],
    handleClick: jest.fn()
  };
  return { ...defaultArgs, ...args };
};

describe("MovieList", () => {
  it("should render an movies", () => {
    const props = renderMovieProps({
      movies: [
        {
          id: 1,
          poster_path: "7GsM4mtM0worCtIVeiQt28HieeN.jpg",
          title: "abc",
          isFav: true,
          watchLater: false
        },
        {
          id: 2,
          poster_path: "7GsM4mtM0worCtIVeiQt28HiedN.jpg",
          title: "xyz",
          isFav: true,
          watchLater: false
        }
      ]
    });
    const tree = renderer.create(<MovieList {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it("should render an one movie if poster path is null", () => {
    const props = renderMovieProps({
      movies: [
        {
          id: 1,
          poster_path: "7GsM4mtM0worCtIVeiQt28HieeN.jpg",
          title: "abc",
          isFav: true,
          watchLater: false
        },
        {
          id: 2,
          poster_path: null,
          title: "xyz",
          isFav: true,
          watchLater: false
        }
      ]
    });
    const tree = renderer.create(<MovieList {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it("should check if fav and Watchlater buttons are added", () => {
    const props = renderMovieProps({
      movies: [
        {
          id: 1,
          poster_path: "7GsM4mtM0worCtIVeiQt28HieeN.jpg",
          title: "abc",
          isFav: true,
          watchLater: false
        }
      ]
    });
    const tree = mount(<MovieList {...props} />);
    expect(tree.find("button").length).toEqual(2);
  });

  it("should highlight fav button if movie is added to fav list", () => {
    const props = renderMovieProps({
      movies: [
        {
          id: 1,
          poster_path: "7GsM4mtM0worCtIVeiQt28HieeN.jpg",
          title: "abc",
          isFav: true,
          watchLater: false
        }
      ]
    });

    const tree = mount(<MovieList {...props} />);
    expect(tree.find(".selected").length).toEqual(1);
  });
});
