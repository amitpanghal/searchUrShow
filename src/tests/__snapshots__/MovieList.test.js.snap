// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList should render an movies 1`] = `
<div
  className="d-flex my-2 flex-wrap movie-container justify-content-center"
>
  <div
    className="movie-div mt-1 mx-3 hover-button"
  >
    <img
      className="img-fluid"
      src="http://image.tmdb.org/t/p/w1547GsM4mtM0worCtIVeiQt28HieeN.jpg"
    />
    <div
      className="fav-div hover-button--on"
    >
      <button
        className="d-flex flex-column m-1 fav-button  selected"
        name="fav"
        onClick={[MockFunction]}
        value={1}
      />
      <button
        className="d-flex flex-column m-1 watchLater-button  "
        name="watchLater"
        onClick={[MockFunction]}
        value={1}
      />
    </div>
  </div>
  <div
    className="movie-div mt-1 mx-3 hover-button"
  >
    <img
      className="img-fluid"
      src="http://image.tmdb.org/t/p/w1547GsM4mtM0worCtIVeiQt28HiedN.jpg"
    />
    <div
      className="fav-div hover-button--on"
    >
      <button
        className="d-flex flex-column m-1 fav-button  selected"
        name="fav"
        onClick={[MockFunction]}
        value={2}
      />
      <button
        className="d-flex flex-column m-1 watchLater-button  "
        name="watchLater"
        onClick={[MockFunction]}
        value={2}
      />
    </div>
  </div>
</div>
`;

exports[`MovieList should render an one movie if poster path is null 1`] = `
<div
  className="d-flex my-2 flex-wrap movie-container justify-content-center"
>
  <div
    className="movie-div mt-1 mx-3 hover-button"
  >
    <img
      className="img-fluid"
      src="http://image.tmdb.org/t/p/w1547GsM4mtM0worCtIVeiQt28HieeN.jpg"
    />
    <div
      className="fav-div hover-button--on"
    >
      <button
        className="d-flex flex-column m-1 fav-button  selected"
        name="fav"
        onClick={[MockFunction]}
        value={1}
      />
      <button
        className="d-flex flex-column m-1 watchLater-button  "
        name="watchLater"
        onClick={[MockFunction]}
        value={1}
      />
    </div>
  </div>
</div>
`;
